### PR TITLE
Make UUID available to users

### PR DESCRIPTION
**Pull request by kaikreuzer**
Implements https://github.com/openhab/openhab-core/issues/2350

Signed-off-by: Kai Kreuzer <kai@openhab.org>

### CODE REVIEW

File: bundles/org.openhab.core.id/src/main/java/org/openhab/core/id/internal/UUIDResource.java [hunk 1]
```diff
@@ -49,7 +49,7 @@
 @JaxrsName(UUIDResource.PATH_UUID)
 @JaxrsApplicationSelect("(" + JaxrsWhiteboardConstants.JAX_RS_NAME + "=" + RESTConstants.JAX_RS_NAME + ")")
 @Path(UUIDResource.PATH_UUID)
-@RolesAllowed({ Role.ADMIN })
+@RolesAllowed({ Role.ADMIN, Role.USER })
 @SecurityRequirement(name = "oauth2", scopes = { "admin" })
```
Comment (cweitkamp) [inline bundles/org.openhab.core.id/src/main/java/org/openhab/core/id/internal/UUIDResource.java:53]: Do we have to remove the `SecurityRequirement` to make this endpoint accessible for users? I could not find any other user defined interface method with it.

Comment (kaikreuzer) [reply bundles/org.openhab.core.id/src/main/java/org/openhab/core/id/internal/UUIDResource.java]: Good point. I actually didn't notice that we have those annotations and am not sure how/if they are considered at all. But it seems to make sense to remove it indeed.

Comment (cweitkamp) [reply bundles/org.openhab.core.id/src/main/java/org/openhab/core/id/internal/UUIDResource.java]: Thanks for removing it. Can you please apply the formatter?

**Comment by pavog** [comment]
Hello everyone!
The `@SecurityRequirement` annotation provides the Open Api specification generator with the information that authentication is required to access this endpoint.
In the end, this leads to:



 ```yaml
 /uuid:
    get:
      tags:
        - uuid
      summary: A unified unique id.
      operationId: getUUID
      responses:
        '200':
          description: OK
          content:
            text/plain:
              schema:
                type: string
```

vs


```yaml
  /uuid:
    get:
      tags:
        - uuid
      summary: A unified unique id.
      operationId: getUUID
      responses:
        '200':
          description: OK
          content:
            text/plain:
              schema:
                type: string
      security:
        - oauth2:
            - admin
```

For more info see [OpenAPI 3 docs -> Authentication and Authorization](https://swagger.io/docs/specification/authentication/).

**Comment by kaikreuzer** [comment]
Thanks for the explanation, @pavog!

File: bundles/org.openhab.core.id/src/main/java/org/openhab/core/id/internal/UUIDResource.java (hunk 1 above)
Comment (kaikreuzer) [reply bundles/org.openhab.core.id/src/main/java/org/openhab/core/id/internal/UUIDResource.java]: Sorry, didn't notice. Done now!

**Review by cweitkamp — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**